src/talk/renderer/setupGlobalInAppTalkLinksHandler.js

```diff
@@ -0,0 +1,77 @@
+/*
+ * @copyright Copyright (c) 2024 Grigorii Shartsev <me@shgk.me>
+ *
+ * @author Grigorii Shartsev <me@shgk.me>
+ *
+ * @license AGPL-3.0-or-later
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import { getBaseUrl } from '@nextcloud/router'
+
+/**
+ * Get Talk route for a given absolute URL or null if it's not a Talk route of the current instance
+ *
+ * @param {import('vue-router')} router - Talk's VueRouter instance
+ * @param {string} url - Absolute URL on this server instance
+ * @return {import('vue-router').Route | null} - Route or null if it's not a Talk route
+ */
+function getTalkRoute(router, url) {
+	const pathWithoutBase = url.replace(getBaseUrl(), '')
+	const pathWithoutIndexPhp = pathWithoutBase.startsWith('/index.php') ? pathWithoutBase.replace('/index.php', '') : pathWithoutBase
+
+	const resolvedRoute = router.resolve(pathWithoutIndexPhp).route
+
+	if (resolvedRoute.matched.length) {
+		return resolvedRoute
+	}
+
+	return null
+}
+
+/**
+ * Intercept into link click handling to navigate via router whenever possible
+ *
+ * @param {import('vue-router')} router - Talk's VueRouter instance
+ */
+export function setupGlobalInAppTalkLinksHandler(router) {
+	document.addEventListener('click', (event) => {
+		const link = event.target.closest('a')
+
+		// Handle only click on links
+		if (!link) {
+			return
+		}
+
+		// Handle only exact clicks
+		if (event.ctrlKey || event.metaKey || event.shiftKey || event.altKey) {
+			return
+		}
+
+		const url = link.href
+
+		// Handle only absolute links to the current server instance. They have a chance to be an internal Talk link
+		if (!url.startsWith(getBaseUrl())) {
```

Comment:
getBaseUrl starts with `http:// https://` protocol, so that would be more performant to check absolute links (if string doesn't match from the first character)

Reply:
Note, `getBaseUrl` on Talk Desktop actually works as `getRootUrl` because we cannot rely on `window.location` as Web page does, and there is no way to force it in `@nextcloud/router` for now.

So it probably should be `getRootUrl` here to be less confusing.

But `includes` cannot be correct anyway. Having the base or root somewhere in a random position in the link doesn't mean it is a Nextcloud internal link.

Reply:
Arr, no, `getRootUrl` originally returns the base path and `getBaseUrl` return an absolute link to the Nextcloud root.